src/autocapture-utils.ts

```diff
@@ -179,6 +221,14 @@ export function shouldCaptureDomEvent(
         curEl = parentNode
     }
 
+    if (!checkIfElementTreePassesElementAllowList(targetElementList, autocaptureConfig)) {
+        return false
+    }
+
+    if (!checkIfElementTreePassesCSSSelectorAllowList(targetElementList, autocaptureConfig)) {
```

Comment:
Based on the description in `types.ts` I would have expected that the elements list check would be different in both cases. Looks like we use `targetElementList` twice. Is it the case that the `targetElementList` changes somehow?

Reply:
can you expand?

we check the same tree every time, applying a different test depending on which allowlist we're looking at, and we don't combine them... e.g. must have img and must have id doesn't mean you only gather clicks from img elements with that id

Reply:
Sorry, maybe I misunderstood here.

From the descriptions in the `types.ts` it read like the element selectors consider children of the allowed element e.g. if an SVG is clicked inside a button we count that as valid. Conversely the CSS selector works if any element in the chain (even a parent of the button) has the specified `id`.
From that I would have expected that the element tree would be manipulated in some way (to remove the two parent elements from consideration in the case of the CSS selector) but it looks like the same chain was passed in both scenarios. Perhaps all of that is covered as part of the checks

Reply:
no, we never consider children 

so, now I'm wondering if the change should go further or the examples should be improved 🤔

Reply:
Gotcha, probably worth getting this out and we can follow up with more if needed